File: solutions/python/dnd-character/1/dnd_character.py
from random import randint

ATTRS = (
    'strength', 'dexterity', 'constitution',
    'intelligence', 'wisdom', 'charisma')

def modifier(score):
    return (score - 10) // 2

class Character:
    def __init__(self):
        for attr in ATTRS:
            setattr(self, attr, self.ability())
        self.hitpoints = 10 + modifier(self.constitution)

    def ability(self):
        return randint(3, 18)

File: solutions/ruby/dnd-character/1/dnd_character.rb
=begin
Write your code for the 'D&D Character' exercise in this file. Make the tests in
`dnd_character_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/dnd-character` directory.
=end

class DndCharacter
  attr_reader :strength, :dexterity, :constitution, :intelligence, :wisdom, :charisma, :hitpoints
  def self.modifier(score)
    (score - 10) / 2
  end

  def initialize
    @strength = DndCharacter.ability
    @dexterity = DndCharacter.ability
    @constitution = DndCharacter.ability
    @intelligence = DndCharacter.ability
    @wisdom = DndCharacter.ability
    @charisma = DndCharacter.ability
    @hitpoints = 10 + DndCharacter.modifier(@constitution)
  end

  private

  def self.ability
    rolls = Array.new(4) { rand(1..6) }
    rolls.sort!.shift
    rolls.sum
  end
end
